### PR TITLE
Newsletters: add #42 (2019-04-16)

### PR DESCRIPTION
**Pull request by harding**
Of the newsletters I've written so far, I think I'm least happy about this one.  I'm sorry it's not better.

Note: the math on the multisig transactions is a bit lazy.  There's at least two places where 1-3 extra bytes are introduced when a field exceeds a certain length; I just used a constant value.  Since the max differences per tx is 6 extra bytes, I think this would be invisible in the chart, but I can fix it if need be.

### CODE REVIEW

File: _posts/en/newsletters/2019-04-16-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+multisig, and native P2WSH multisig for all the possible k-of-n multisig
+combinations (up to the maximum 15-of-15 supported by legacy P2SH).  We
+can see that switching to P2SH-P2WSH brings large savings.
+Further switching to native P2WSH brings an additional constant-sized
+savings (about 35 bytes) that can increase savings by up to 15% for
+small multisigs (but which is marginal for large multisigs).
+
+![Plot of multisig transaction sizes with P2SH, P2SH-P2WSH, and P2WSH](/img/posts/2019-04-segwit-multisig-size.png)
+
+This accounts for almost all scripts being used with legacy and wrapped
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-04-16-newsletter.md]: As discussed in IRC, I think these last two paragraphs are a little hard to parse. My suggestion is to cut them right back to something simple like "If more users transition to using the more efficient transaction types, total blockspace demand decreases and everyone's fees should be lower." Perhaps that's making it too simplistic though?

File: _posts/en/newsletters/2019-04-16-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+and the number of public keys (n).  So, for simplicity's sake, we'll
+just plot the sizes of legacy P2SH-multisig, wrapped P2SH-P2WSH
+multisig, and native P2WSH multisig for all the possible k-of-n multisig
+combinations (up to the maximum 15-of-15 supported by legacy P2SH).  We
+can see that switching to P2SH-P2WSH brings large savings.
+Further switching to native P2WSH brings an additional constant-sized
+savings (about 35 bytes) that can increase savings by up to 15% for
+small multisigs (but which is marginal for large multisigs).
+
+![Plot of multisig transaction sizes with P2SH, P2SH-P2WSH, and P2WSH](/img/posts/2019-04-segwit-multisig-size.png)
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-04-16-newsletter.md]: I like this graph, but the P2SH-P2WSH plots are almost impossible to see. We could experiment with different layouts, or perhaps have two graphs. One for 1-of-n multisigs (for 1 <= n <= 15) and one for a k-of-15 graph (for 1 <= k <= 15).

File: _posts/en/newsletters/2019-04-16-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  addresses is not changed.
+
+- [C-Lightning #2506][] adds a `min-capacity-sat` configuration
+  parameter to reject channel open requests below a certain value.  This
+  replaces a hardcoded minimum of 0.00001 BTC previously in the code.
+
+- [LND #2933][] adds a document describing LND's current [backup and recovery
+  options][lnd recover].
+
+- [LND #2908][] logs the reason the node forced-closed one of its
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-04-16-newsletter.md]: Generally, I think that logging PRs don't make the cut for 'interesting for our readers'. Is there a reason that this is different?

Comment (harding) [reply _posts/en/newsletters/2019-04-16-newsletter.md]: I think that's a good policy, and I'll remove this.  My motivation was purely, "oh, I'd use that".

**Comment by harding** [comment]
Added a few separate small commits that I think address @jnewbery's feedback (thanks!).  They're separate commits to make them easy to revert if desired.  Please feel free to make any last-minute edits you think are necessary.

File: _posts/en/newsletters/2019-04-16-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+- Spending P2WPKH output has a size of 141 vbytes (16% savings vs
+  P2SH-P2WPKH or 35% vs P2PKH)
+
+To compare simple multisig transactions (those that just use a single
+`OP_CHECKMULTSIG` opcode), things get more complex because k-of-n
+multisig inputs vary in size depending on the number of signatures (k)
+and the number of public keys (n).  So, for simplicity's sake, we'll
+just plot the sizes of legacy P2SH-multisig compared to wrapped P2SH-P2WSH
+multisig (up to the maximum 15-of-15 supported by legacy P2SH).  We can
+see that switching to P2SH-P2WSH brings large savings.
```
Comment (moneyball) [inline _posts/en/newsletters/2019-04-16-newsletter.md]: Should we quantify large? For the other cases we show #s, and it is hard to know the % savings by glancing at the graph.

File: _posts/en/newsletters/2019-04-16-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+fork lock-in on 24 August 2019, the Optech Newsletter will contain this
+weekly section that provides information to help developers and
+organizations implement bech32 sending support---the ability to pay
+native segwit addresses.  This [doesn't require implementing
+segwit][bech32 easy] yourself, but it does allow the people you pay to
+access all of segwit's multiple benefits.*
+
+One reason your users and customers may want you to implement bech32
+sending support is because it'll allow the receivers of those payments
+to save on fees when they re-spend that money.  This week, we'll look at
```
Comment (moneyball) [inline _posts/en/newsletters/2019-04-16-newsletter.md]: Totally optional but could perhaps mention that there are use cases where you spend to yourself (ie UTXO consolidation) in which you, the sender, are the recipient of this benefit.

**Comment by moneyball** [comment]
tACK 584f1b1740c81038dc59c26b767acf87a20d1ae6

left a couple comments but approve as is

File: _posts/en/newsletters/2019-04-16-newsletter.md (hunk 5 above)
Comment (harding) [reply _posts/en/newsletters/2019-04-16-newsletter.md]: This is a good idea, but I looked at it and think it weighs down the introduction too much.

**Comment by harding** [comment]
Pushed a change for one of @moneyball's suggestions (thanks!).  Should be good to go.

**Comment by jnewbery** [comment]
ACK 32a8c9c41a4e206455851181d7c6ba66fc799e20.

Time to squash, merge and :ship:

**Comment by jnewbery** [comment]
Force-pushed a squashed branch. No changes to content: https://github.com/bitcoinops/bitcoinops.github.io/compare/32a8c9c41a4e206455851181d7c6ba66fc799e20..7164b52fb46b9ac5004686376ed9496f3294d983
